### PR TITLE
update validateFrameAction to take in channelFollowContext, updates fetchTrendingChannels return type

### PR DESCRIPTION
**Pull request by MrKevinOConnell**
NOTE: relies on https://github.com/neynarxyz/OAS/pull/83 and https://github.com/neynarxyz/OAS/pull/84

updated:
validateFrameAction -> https://docs.neynar.com/reference/validate-frame
adds channelFollowContext as an option.

misc:
fetchTrendingChannels ->  https://docs.neynar.com/reference/trending-channels
now returns TrendingChannelResponse 
lookupCastConversation -> https://docs.neynar.com/reference/cast-conversation
Type is wrapped around an object that contains conversation

### CODE REVIEW

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]
@MrKevinOConnell Please update the JSDocs with the updated limit for `fetchAllNotifications`, `fetchChannelNotificationsForUser`, `fetchNotificationsByParentUrlForUser`. The rest looks good.
Also please sync-up with the OAS changes it'll remove allof.

**Comment by MrKevinOConnell** [comment]
@Shreyaschorge seems like the updated limits are correct

**Comment by Shreyaschorge** [comment]
> OAS

@MrKevinOConnell I meant it needs to be updated in methods documentation where we give an example of how to use a method.

File: src/neynar-api/v2/client.ts [hunk 1]
```diff
@@ -1558,17 +1559,17 @@ export class NeynarV2APIClient {
    *
    * @param {number} fid - The FID of the user whose notifications are being fetched.
    * @param {Object} [options] - Optional parameters to tailor the request.
-   * @param {number} [options.limit] - Number of results to retrieve (default 25, max 50).
+   * @param {number} [options.limit=25] - Number of results to retrieve (default 15, max 25).
```
Comment (Shreyaschorge) [inline src/neynar-api/v2/client.ts:1562]: One last change, then its good to merge.
default limit is 15

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]

**Review by Shreyaschorge — APPROVED** [review]

**Pull request merged**